misc/server/Dockerfile

```diff
@@ -1,8 +1,12 @@
-FROM rust:1.66-bullseye as builder
+FROM rust:1.72-bullseye as builder
 WORKDIR /usr/src/rust-libp2p-server
 
 # Run with access to the target cache to speed up builds
 WORKDIR /workspace
+
+RUN --mount=type=cache,target=/usr/local/cargo/registry \
+    cargo install --root /usr/local libp2p-lookup  --version 0.6.4
```

Comment:
I believe it is best practice to use `--locked` when installing binaries, otherwise `cargo install` will not use the lockfile that is stored in the crate.

Reply:
Good call, thanks!